### DISCUSSION TITLE
how to migrate express graphql to graphql-http with request

### DISCUSSION THREAD

**Discussion by blee-deloitte** [graphql/graphql-http#85] [Q&A]
Hi, i tried to look for how to migrate express-graphql to graphql-http with req function but i couldn't find anything in github or online.
here is the example: 

**graphqlhttp version** 
app.use('/graphql', graphqlHTTP(req => {context: req.headers,}) 

**i tried to do the same method for createHandler but its not reading the function.** 
app.use('/graphql', createHandler(req => {context: req.headers})

Also, does anyone knows if graphql-http can use **const  NoIntrospection  = require('graphql-disable-introspection');** for validationRules in express-graphql?

Thank you in advance

**Comment by enisdenjo** [comment]
The `createHandler` for express does not accept a function as an argument. You can find the information about [it in the documentation](https://github.com/graphql/graphql-http/blob/main/docs/interfaces/handler.HandlerOptions.md).

```diff
- app.use('/graphql', createHandler(req => {context: req.headers})
+ app.use('/graphql', createHandler({context: req => req.headers})
```

Furthermore, the handler accepts [a `validationRules` option](https://github.com/graphql/graphql-http/blob/main/docs/interfaces/handler.HandlerOptions.md#validationrules) where you can pass additional validation rules.

```js
const NoIntrospection = require('graphql-disable-introspection');

app.use('/graphql', createHandler({validationRules: [NoIntrospection]})
```
